test/functional/Java8andUp/src/org/openj9/test/constantpool/ClassInitTest.java

```diff
@@ -0,0 +1,311 @@
+package org.openj9.test.constantpool;
+
+/*******************************************************************************
+ * Copyright (c) 2021, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Member;
+import java.lang.reflect.Method;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.function.Consumer;
+
+@Test(groups = { "level.sanity" })
+public class ClassInitTest {
+	static Method getConstantPool;
+	static Object javaLangAccess;
+
+	static Method getSize;
+	static Method getClassAtIfLoaded;
+	static Method getClassAt;
+	static Method getMethodAtIfLoaded;
+	static Method getMethodAt;
+	static Method getFieldAtIfLoaded;
+	static Method getFieldAt;
+	static boolean clinitInvoked = false;
+
+	public ClassInitTest() throws Throwable {
+		String version = System.getProperty("java.vm.specification.version");
+
+		Class SharedSecrets;
+		Class JavaLangAccess;
+		Class ConstantPool;
+
+		if (Double.parseDouble(version) == 1.8) {
```

Comment:
@llxia Are you aware of a standard way to do these version checks in tests?

Reply:
I recommend `org.openj9.test.util. VersionCheck`
https://github.com/eclipse-openj9/openj9/blob/bad687ee7ae3ff98cf547882ba4abb77e19f1b69/test/functional/TestUtilities/src/org/openj9/test/util/VersionCheck.java#L53-L58

Reply:
Changed to use VersionCheck. Tested locally on 8, 11, and 17 and passed